### ISSUE TITLE
Emulating the stairstepping of the original DACs

### ISSUE THREAD

**Issue by nikitalita** [nukeykt/Nuked-SC55#50]
This is something that may not be able to be satisfactorily accomplished with a software implementation, and likely not relevant to anyone but the nerdiest of audiophile nerds, but I thought I'd mention it:

The original DACs for these devices (NEC UPD63200 and the NEC UPD6376) did stair-stepping for the output:

![image](https://github.com/nukeykt/Nuked-SC55/assets/69168929/98250747-019f-478c-be53-ac1afbac33c1)
(logic analyzer traces from the DAC on a SC-55mk2; digital PCM input on top, DAC analog output on the bottom)

This is unlike modern DACs, which interpolate between samples for nice smooth waveforms. So, the audio output of a real SC-55 would sound a bit crunchier than the audio output of this emulator going through a modern DAC.

One option that I can think of that might be able to be done in a software implementation to emulate this behavior would be to resample to a much higher sample rate (like 384k) and ensure that the samples are just straight up copied instead of interpolated.
Of course, this would only be an option if the user had a sound card that supported high sample rates, and we would still have to deal with the weird sample rate (66.207khz) of the mk2 by interpolating the bare minimum amount of samples to get it to an integer ratio with the higher sample rate.
Anyone else have ideas or suggestions on this?

**Comment by Axis4s** [comment]
While having accurate emulation is good, I do think some things should be left behind in the original hardware

**Comment by michael02022** [comment]
I couldn't say accurate emulation is bad. But I have to differ this suggestion because accurate DAC is often done for very cheap gear (like videogame hardware) which such low-end DACs or specifications defines the music produced in them (in other words, it has a musical/sound design purpose), an musical example of this would be Roland D-50, which the low quality of it made the presets specifically designed based on those low specs. (https://youtu.be/nqiMU8p4PH4 https://youtu.be/VggsB5eZ0oM)

This is not the case for SoundCanvas due to being a "prosumer" product (it was conceived as a professional musical instrument, and that in later revisions, they were increasing the quality of the DAC because it is for music production).

You can see this review from 1991 from a magazine dedicated to music gear, which says:

> WHAT'S NOT TO LIKE?
> Are there any reasons not to like the Sound Canvas? Scant few. The unit is a little noisy, with a white-noise tail plus a very small bit of 'grunge' at the very end which hangs on the end of every sound. 
https://www.muzines.co.uk/articles/roland-sc55-sound-canvas/7496

**Comment by Burrito78** [comment]
I bet that in the hardware there is some analogue output circuits to smooth this out nearly 100%.

**Comment by nikitalita** [comment]
> I bet that in the hardware there is some analogue output circuits to smooth this out nearly 100%.

perhaps; I haven't tested the audio output from the RCA jacks with a scope yet. Here's the schematic for the analog circuit attached to the DAC (far left, IC9, UPD63200)
<img width="1019" alt="image" src="https://github.com/nukeykt/Nuked-SC55/assets/69168929/2d8766ff-c1fb-4011-ad69-82c7147f91cb">

**Comment by matt713** [comment]
IMHO, such thing is outside the scope and most importantly the main goal of this project, which aims to achieve bit-perfect output of the digital circuit and doing so with using hard-evidence based on decapping of the PCM chip, otherwise it can be done based on educated assumption what the PCM chip is supposed to do (similarly how MAME Yamaha SWP00 emulator is done or even what the old "emusc" project is trying to do). so, adding on top of that what the DAC chip is supposed to do (and even the analog circuit after that) based on educated assumptions defeats the whole purpose of the PCM chip decapping.

**Comment by Axis4s** [comment]
> > IMHO, such thing is outside the scope and most importantly the main goal of this project, which aims to achieve bit-perfect output of the digital circuit and doing so with using hard-evidence based on decapping of the PCM chip, otherwise it can be done based on educated assumption what the PCM chip is supposed to do (similarly how MAME Yamaha SWP00 emulator is done or even what the old "emusc" project is trying to do). so, adding on top of that what the DAC chip is supposed to do (and even the analog circuit after that) based on educated assumptions defeats the whole purpose of the PCM chip decapping.
> 
> Because of this, I request this issue to be closed.

agreed

**Comment by michael02022** [comment]
> IMHO, such thing is outside the scope and most importantly the main goal of this project, which aims to achieve bit-perfect output of the digital circuit and doing so with using hard-evidence based on decapping of the PCM chip, otherwise it can be done based on educated assumption what the PCM chip is supposed to do (similarly how MAME Yamaha SWP00 emulator is done or even what the old "emusc" project is trying to do). so, adding on top of that what the DAC chip is supposed to do (and even the analog circuit after that) based on educated assumptions defeats the whole purpose of the PCM chip decapping.

Pretty much the issue here is we are emulating a device that is meant to be profesional, unlike other emulations.

We should think the Nuked SC-55 as a SC-55 with digital audio output (and that's what people actually want, people wanted the SC-55 to sound high quality, CD quality level music from their devices).

If people want to try to emulate the DAC, I have no problems with it, is just this is kind very niche and doesn't make much sense imho.

**Comment by michael02022** [comment]
It reminds me when people went from analog to digital audio recording in early 70s, and they seen that as a positive thing.

It would be something like this with this emulator, the fact we can output totally clean audio is a positive thing, not something to be "fixed" as such in most cases.

**Comment by matt713** [comment]
> If people want to try to emulate the DAC, I have no problems with it

I don't have problem with it in general, but my point is Nuked-* emulators are always based on decapped chips and thus on hard-evidence how the emulated chip(s) exactly work. so, even as approach current DAC proposal is not conceptually compatible with this project - I mean, at least I make huge difference between "educated guesses" how it's supposed to work and derived work from a decap chip. So, if someone wants to decap the DAC and make code that exactly emulates its work, then it can be added to this project as an option, but otherwise, if it will be work based on educated guesses how the DAC is supposed to work - then it's just not its place to be in Nuked-* emulator. of course, everyone is free to span the open-source code and make a branch with DAC guesstimation there.

**Comment by michael02022** [comment]
> > If people want to try to emulate the DAC, I have no problems with it
> 
> I don't have problem with it in general, but my point is Nuked-* emulators are always based on decapped chips and thus on hard-evidence how the emulated chip(s) exactly work. so, even as approach current DAC proposal is not conceptually compatible with this project
> 

Yeah, I understand the point, but also I wanted to explore the issue based on a musical perspective too.

**Comment by Karmeck** [comment]
There is one thing to be perfectly emulated and an other to emulate perfectly authentic. Like waterfalls in sonic games are made in a way to look good on crts, the midis we listen to are made on this, imperfect, analog, hardware. So to make it perfect instead of authentic changes conditions in a way that the midis might not sound "as the composer intended".

I can agree though that it might be out if scoop. 

I also fail to find where nukeykt declares is intensions for this project. Good that we have people here that can speak for them.

**Comment by markanini** [comment]
1. The audio is already a bit crunchy like the original units were. You can hear it when listening through good headphones. I'd say that a lot of the character of the original has been preserved. This would be a result of mid 90s digital signal processing, an issue of managing limited processing resources, not DAC related.

2. The idea that actual stair stepping was in the analog output is unlikely. Something in this analysis is wrong, the output character would be far different that what everyone knows as the SC55 sound if it were, and given the available tech in the mid-90s, DACs that were able to fully reconstruct a continuous waveforms were readily available, even cheaply.

**Comment by giulioz** [comment]
> This would be a result of mid 90s digital signal processing, an issue of managing limited processing resources, not DAC related.

I very much agree with that, I think that when people are discussing about how "a synth sounds good because of the DAC" they are most of the times wrong, as DACs are always designed to be as accurate as possible.

That said, I'm measuring the output of the JV880 vs the emulated one (I don't have an SC55 but they are very close) and I can hear some difference. Not really stair stepping, but the frequency response is different probably due to the design of the output amplifier and LPF filter.

I tried to check the difference with a matching EQ in Logic Pro and I got the following results:

<img width="1428" alt="image" src="https://github.com/nukeykt/Nuked-SC55/assets/1353142/d3f1ff34-b5ba-439b-87fb-d889841d53ec">

You can see how the real unit has some level of analog EQ in the output stage, mostly increasing the low and high end. Reproducing that in software could be very easy and could make the emulator sound even closer to the original, especially if it becomes a feature that the user can opt out from.

**Comment by nikitalita** [comment]
> I would be very surprised if the output stage LPF of the SC55 doesn't de-alias those steps completely tbh, so if you can please measure from the RCA output, curious about the result!

It doesn't!

Hooked it up to my digital scope, captured at 50Mhz:
![image](https://github.com/nukeykt/Nuked-SC55/assets/69168929/2946ab98-6e44-4f32-a633-44f0cc8fbf89)
![image](https://github.com/nukeykt/Nuked-SC55/assets/69168929/6deb0e30-7aee-49f5-a77c-58bb89e9746c)

**Comment by nikitalita** [comment]
Here's the output of the sine wave from the sound test, which makes it a bit more obvious:
![image](https://github.com/nukeykt/Nuked-SC55/assets/69168929/0b7d9658-f19c-4c06-a8d9-49c3eb202975)

I'd do a comparison vs. Nuked SC-55 for the sine wave, but I can't seem to get it into test mode.

**Comment by michael02022** [comment]
> If people want to try to emulate the DAC, I have no problems with it, is just this is kind very niche and doesn't make much sense in practice imho.

Just to add something about this. I made a very vulgar test with a hardware SC-55mkII raw recorded midi file vs Nuked SC-55 and switched between them, blind.

It's basically a WAV vs OGG case (or whatever high quality lossy compression format), the differences are entirely psychological, to the point that you have to use absurdly specialized equipment in order to be able to pick up the differences.

> There is one thing to be perfectly emulated and an other to emulate perfectly authentic. Like waterfalls in sonic games are made in a way to look good on crts, the midis we listen to are made on this, imperfect, analog, hardware. So to make it perfect instead of authentic changes conditions in a way that the midis might not sound "as the composer intended".

This is false, there's no equivalent of "waterfall effect" in the SC-55 because **all of it is digital and played through digital gear** pretty much the CC values and the way those are synthesized are the same. The way the patches were programmed are the same, and so on. This argument could be applied to the official Roland's plugins instead which has a worse emulation in comparison which indeed changes the intention of composer, but not for the case of Nuked SC-55. The intention of the composer are in the digital algorithms of the SC-55, not in the DACs, this is not a videogame chip where composers made some audio tricks to enhance certain things (4-bit PCM in SID, ladder DAC effect in Genesis' FM chip, ...)

And as I said earlier, the audio differences between hardware and emulated are practically unaudible, and of course, the composers cannot heard these effects (because these composers used high-end gear to playback the SC-55) and therefore the DAC effect was totally ignored for their compositions because in a musician perspective it was an imperfection of the device and not a feature.

**Comment by zaphod77** [comment]
i'd think you'd be able to hear a difference in the sine wave tests between real hardware and the emulator.   i agree it would probably be hard to hear a difference for most stuff, but there's a certain character to the "charang" at pretty low pitches that i hear on every version of VSC, sound canvas va, and the ms samples that seems to be missing in nuked-sc55. and i made a dance cover midi file of automatic lover that actually depended on this, using the charang patch to do an imitation of the words "automatic lover" that you could kinda hear. i of couse don't have a real sc55 to play the midi back on.

was the charang sample updated at some point between sc55 and sc8820? I know some of the samples from the MS set were lifted from sc-88 and not sc-55, but i don't know which ones.

**Comment by michael02022** [comment]
> i'd think you'd be able to hear a difference in the sine wave tests between real hardware and the emulator. i agree it would probably be hard to hear a difference for most stuff, but there's a certain character to the "charang" at pretty low pitches that i hear on every version of VSC, sound canvas va, and the ms samples that seems to be missing in nuked-sc55. and i made a dance cover midi file of automatic lover that actually depended on this, using the charang patch to do an imitation of the words "automatic lover" that you could kinda hear. i of couse don't have a real sc55 to play the midi back on.
> 
> was the charang sample updated at some point between sc55 and sc8820? I know some of the samples from the MS set were lifted from sc-88 and not sc-55, but i don't know which ones.

SC-55 patches are very different in terms of synthesis in later models, and if you're comparing it with the SC8820, well, SC8820 is so different to the original SC-55 even if are "the same patches" we have this issue people that are into sample finding.

**Comment by michael02022** [comment]
> actually, i tried again, and sound canvas va in sc-55 mode pretty much sounds the same as nuked with my midi after all.

Set your SCVA in "SC-55 mode" to preset 49 (Strings) and you will find a surprise || is not even the same instrument as the original SC-55 ||

**Comment by zaphod77** [comment]
i know there are differences between the samples even in compatibility modes, but my point is that this wasn't my problem after all.  just that the soundfonts and earlier softsynths were wrong, and i would need to add some TVF NRPN to actually fix it on  a real SC.

**Comment by johnnovak** [comment]
Nice to see that after like 5 pages of detailed technical speculations some people _actually_ listened to the analog output with their ears too 😏 

The analog stages or whatever extra circuitry is after the digital stages definitely impact the sound you actually hear coming out of the unit. I could hear this difference when comparing my SC-55 mk2 real hardware recordings with the SCVA, and the results are very similar with the NukedSC55.

All my recording are available, check links in my article:
https://blog.johnnovak.net/2023/03/05/grand-ms-dos-gaming-general-midi-showdown/

The differences are quite audible in _full spectrum_, busy audio with lots of bass, high end, and lots of sharp transients, aka drums. The Shadow Warrior soundtracks and the System Shock intro tune are perfect examples. So yeah, hard-hitting rock and electronic styled music is where you hear the analog ouput being fuller, better, more exciting.

With orchestral tracks, sparse tracks with acoustic instruments, or anithing without deep bass and drums, you won't hear much of a difference. But on rock/electro the bass and the spikey transients cause some pleasant subtle distortion in the analog stages. You can hear this as extra fatness, sparkle, and "more 3D-ness" because of the added harmonics caused by these subtle distortions.

Do note my overdrive tests in my article. E.g. the Shadow Warrior soundtrack starts clipping the analog stages with the output volume set to around 10 o'clock. This is clearly visible on the recorded waveforms. You can actually turn the volume up to 3 o'clock and _it still sounds very good, even better_! But the waveform starts looking like a fat sausage with the spikey transients chopped off. That's analog magic for ya 😎 Sound engineers all over the world have been using nalog saturators to make the signal more interesting since the 60s. Digital recreation of these analog saturators are highly sought after and very common in digital music production (Google "saturator plugin daw"l.

The real sound of the unit is _with_ these characteristics. But accurately emulating these are hard, super expensive, and a bit pointless because you xan approximate is with some multiband saturator/distortion plugin and add some overdrive to the lower bands.

People designed audio device as a system in the 80s/90s, and the analog characteristics were taken into account. The digital output of many 89s/90s synths sound cold and sterile versus their analog out. Many studio guys prefer the analog out, thst's the real sound of the device.

Anyway, that's probably out of scope for the project at this stage. Accurate analog circuit emulation would be way overkill, but slapping on a multiband saturator with some nice presets would do the job

**Comment by giulioz** [comment]
@johnnovak agreed, I could hear a clear difference and even measure it as proved with the frequency response chart I posted before. Do you think that an approximation with an impulse response could yield some results? The idea would be to create it playing a sine sweep through the sound chip in the real device recording the audio output.

**Comment by johnnovak** [comment]
> It reminds me when people went from analog to digital audio recording in early 70s, and they seen that as a positive thing.

You interviewed the wrong people then 😄 

People were initially _shocked as fuck_ how sterile and cold 100% early digital productions sounded? Remember AAD, ADD, and DDD? Maybe DDD was OK for classical, but definitely not anything with bass and drums in it.

People are still searching for the "holy grail" of recreating the analog magic in the digital realm to this day.

Digital is the best reproduction and archival medium (probably), but that is its weakness for creative work: it lacks "colour" and "artifacts" that makes analog synths and audio gear so exciting to use.

**Comment by markanini** [comment]
I would be harder for me to accept two statements to be true at the same time.
"A mid-1990s Roland digital synth module has warmth and character in it's outputs akin to a current day saturator plugin."
"Digital is cold and sterile"

**Comment by johnnovak** [comment]
> I would be harder for me to accept two statements to be true at the same time. "A mid-1990s Roland digital synth module has warmth and character in it's outputs akin to a current day saturator plugin." "Digital is cold and sterile"

Trust your ears 😎

**Comment by matt713** [comment]
ok, I want to ask - In your opinion (the people commenting here) what is the "authentic" or "true" or whatever SC-55 analog sound? I am asking this, considering Roland used the same Digital board (RLP-XYZT, e.g. RLP-3037) in literally countless products with different DACs and designs: CM-300, CM-500, MT-200, JW-50 just to name a few and then - are they all in your eyes (or should I say "ears") a "lesser" SC-55 GS synthesizer?! Also, CM ("Computer Module") series of devices are reported to be the only giving proper compatibility with number of games, at least based on the following posts (I haven't tested/investigated it myself):

https://www.vogons.org/viewtopic.php?p=1110160#p1110160

https://www.vogons.org/viewtopic.php?p=1092925#p1092925

I don't know about CM-500, but CM-300 is using Burr-Brown DAC - I guess then even it has excellent DOS/Games compatibility according to the discussion here it's worse SC-55/GS Synth, because it doesn't use "stair-stepping" NEC UPD63200 or NEC UPD6376 DAC as some of the other SC-55 units. Do I understand right, because at least I am lost in the exact purpose of this discussion?

**Comment by giulioz** [comment]
@matt713 

No, it's not about stair stepping at all, which is instead only the product of a wrong measurement. What is being discussed here is the analog output stage of the different devices.

It's not about the SC-55 specifically. Every single Roland device that has an analog output has at least a DAC, an amp and a series of filters/EQ, from the MT-32 to the SC-series to the CM-series, ALL of them. The goal of those components is to transform the digital bits to a voltage audio signal that can be put into amplified speakers. While the digital emulation does 95% of the job, there is still a tiny bit of sound character that gets lost in emulation because of those missing components.

Imagine emulating a Nintendo NES perfectly but showing the digital video output only. It would be technically accurate, but it would lose all the smoothing effect of the composite video output and the CRT-look, which is the thing that the developer expected to have in the final result.
What we are talking about here though is audio: aliasing in high frequencies, harmonic distortion, background noise. Having those would make the emulation "less precise", but much closer to the experience that you would get with the real deal.

The discussion here is: should this emulator only precisely reproduce the digital behavior, or should it also reproduce those effects? And if so, what's the depth in which this should be done? Because compared to the digital part, for this there is absolutely no "right" way to do it. A simple EQ could be implemented or it could go as far as emulating every transistor, resistance and capacitor (not happening lol).

And this would not be a first: the MUNT emulator for the MT-32 takes care of analog emulation as well, which is approximated using a FIR filter: https://github.com/munt/munt/blob/master/mt32emu/src/Analog.cpp

**Comment by johnnovak** [comment]
> ok, I want to ask - In your opinion (the people commenting here) what is the "authentic" or "true" or whatever SC-55 analog sound? 

Did you listen to my recordings? Do you hear the difference? That's my answer—listen to the recordings, and hear it for yourself 😄

**Comment by johnnovak** [comment]
> The discussion here is: should this emulator only precisely reproduce the digital behavior, or should it also reproduce those effects? And if so, what's the depth in which this should be done? Because compared to the digital part, for this there is absolutely no "right" way to do it. A simple EQ could be implemented or it could go as far as emulating every transistor, resistance and capacitor (not happening lol).


> And this would not be a first: the MUNT emulator for the MT-32 takes care of analog emulation as well, which is approximated using a FIR filter: https://github.com/munt/munt/blob/master/mt32emu/src/Analog.cpp

And so does reSIDfp, various Amiga emulators, and my own analog output filter emulations I added to DOSBox Staging. All attempt to emulate analog filtering and distortions to various degrees and the pecularities of the DAC / analog stage. E.g., the Amiga outputs basically square waves at MHz rates (I think), then they slapped some crude lowpass filters at the end to smooth out the output somewhat. It's aliasing city but _that is the sound_ of the Amiga! That accurate "DAC" emulation in WinUAE _really_ gets me in the mood given I spent hundreds of hours writing music on the Amiga. Digital-only recreations just don't cut it. If you play the PCM samples with "high-quality" inteprolation methods, that's not the Amiga sound but something completely different. The character is in that low-fi crud.

And so on.

**Comment by nikitalita** [comment]
>
> No, it's not about stair stepping at all, which is instead only the product of a wrong measurement. What is being discussed here is the analog output stage of the different devices.

How is it "wrong"? Did you not see the capture from the RCA jacks I did??

https://github.com/nukeykt/Nuked-SC55/issues/50#issuecomment-2067927736

If you mean that "the other analog effects have a greater influence on the sound than the stairstepping", that's fine, but I've shown that this does in fact show up in the output.

**Comment by johnnovak** [comment]
> 2\. So you would have to argue that Roland chose ultra budget components for a professional device which is unlikely.

Just saw this. The SC-55 was never intended to be a professional device. I was intended for gamers as a low-cost entry-level device, and yeah, at least the mk1 is full of cheap hacks. I noticed these quantisation noise tails in my recordings too, and the unit is generally very noisy.

1991 review:
https://www.muzines.co.uk/articles/roland-sc55-sound-canvas/7496

> Are there any reasons not to like the Sound Canvas? Scant few. The unit is a little noisy, with a white-noise tail plus a very small bit of 'grunge' at the very end which hangs on the end of every sound. The length of the tail varies from about three to six seconds, and varies among the different instruments, but it is not in any way (as far as I could tell) related to an instrument's envelope. It is not velocity related, so quiet passages end up with proportionally more noise than loud ones, nor is it pitch related.

>Even when it is present, it is more curious than objectionable, but it makes the Sound Canvas unsuitable for situations when a large dynamic range and low noise floor are called for — you would not want to cut a solo piano CD with this box, even though the grand piano sound is quite respectable.

**Comment by Grieferus** [comment]
> > 2. So you would have to argue that Roland chose ultra budget components for a professional device which is unlikely.
> 
> Just saw this. The SC-55 was never intended to be a professional device. I was intended for gamers as a low-cost entry-level device, and yeah, at least the mk1 is full of cheap hacks. I noticed these quantisation noise tails in my recordings too, and the unit is generally very noisy.
> 
> 1991 review: https://www.muzines.co.uk/articles/roland-sc55-sound-canvas/7496
> 
> > Are there any reasons not to like the Sound Canvas? Scant few. The unit is a little noisy, with a white-noise tail plus a very small bit of 'grunge' at the very end which hangs on the end of every sound. The length of the tail varies from about three to six seconds, and varies among the different instruments, but it is not in any way (as far as I could tell) related to an instrument's envelope. It is not velocity related, so quiet passages end up with proportionally more noise than loud ones, nor is it pitch related.
> 
> > Even when it is present, it is more curious than objectionable, but it makes the Sound Canvas unsuitable for situations when a large dynamic range and low noise floor are called for — you would not want to cut a solo piano CD with this box, even though the grand piano sound is quite respectable.

I wouldn't say original SC-55 was intended for gamers. Rather, it was aimed at amateur MIDI composers, because it contains lots of buttons that would be necessary only if you're composing MIDIs for games. However, lots of playback-only versions, such as SC-55ST and SC-7 were intended for gamers.

**Comment by johnnovak** [comment]
> I wouldn't say original SC-55 was intended for gamers. Rather, it was aimed at amateur MIDI composers, because it contains lots of buttons that would be necessary only if you're composing MIDIs for games. However, lots of playback-only versions, such as SC-55ST and SC-7 were intended for gamers.

My bad, I think you're right, that's more accurate, agreed. I happen to have both the SC-55 mk1 and the SC-55st, and yeah.

**Comment by nikitalita** [comment]
The SC-55 was STEEP for anyone at the time; it retailed at $695. The reduced versions (SC-50, SC-55ST) went for around $595. Gamers were not buying these things. Professionals on a budget were buying these things.

**Comment by johnnovak** [comment]
> The SC-55 was STEEP for anyone at the time; it retailed at $695 ($1,593 inflation adjusted). The reduced versions (SC-50, SC-55ST) went for around $595. Gamers were not buying these things. Professionals on a budget were buying these things.

They were, just the rich ones. Same story as for the MT-32, the prices were comparable.

Remember, most of the Sierra adventure game fanbase was 30-40 year old loaded professionals in the US, they could afford this stuff.

**Comment by johnnovak** [comment]
People interested in recreating the actual analog sound of the unit vs the digital only parts, please read the relevant parts of my article where I compare the hardware SC-55 and Yamaha MU80 units and their software recreations (NukedSC55 wasn't available then, but the SCVA findings fully apply to it regarding the missing analog character).

Note that I've uploaded FLAC recordings of 46 DOS soundtracks on all these devices. Links and details in the blog post, I recommend reading the whole thing and listening to the recordings.

**Roland SC-55 vs Roland Sound Canvas VA**
https://blog.johnnovak.net/2023/03/05/grand-ms-dos-gaming-general-midi-showdown/#roland-sc-55-vs-roland-sound-canvas-va

**Footnote about SPDIF vs analog outs**
https://blog.johnnovak.net/2023/03/05/grand-ms-dos-gaming-general-midi-showdown/#fn:2

**Yamaha MU80 vs Yamaha S-YXG50**
https://blog.johnnovak.net/2023/03/05/grand-ms-dos-gaming-general-midi-showdown/#yamaha-mu80-vs-yamaha-s-yxg50


https://blog.johnnovak.net/2023/03/05/grand-ms-dos-gaming-general-midi-showdown/#fn:2

**Comment by markanini** [comment]
If an apples to apples comparison is desired, assuming that you don't have SC-55 hardware on hand. Here are analogue SC-55 captures of some popular 1990s game titles. It can be compared to Nuked SC55s output playing source midi files, for instance sourced from here https://www.vgmpf.com

Johnnovak's selection of captures were before Nuked SC-55 was released, thus you will need to take the extra steps above If you want to tell by your own ears about Nuked SC-55 vs hardware SC-55 specifically.

**Comment by Karmeck** [comment]
> If an apples to apples comparison is desired, assuming that you don't have SC-55 hardware on hand. Here are analogue SC-55 captures of some popular 1990s game titles. It can be compared to Nuked SC55s output playing source midi files, for instance sourced from here https://www.vgmpf.com
> 
> Johnnovak's selection of captures were before Nuked SC-55 was released, thus you will need to take the extra steps above If you want to tell by your own ears about Nuked SC-55 vs hardware SC-55 specifically.

I'm confused, what is wrong with the recordings johnnovak made? The hardware ones that is.

**Comment by johnnovak** [comment]
> I'm confused, what is wrong with the recordings johnnovak made? The hardware ones that is.

Nothing. I haven't recorded the NukedSC55 version yet, that is all.

**Comment by Karmeck** [comment]
> > I'm confused, what is wrong with the recordings johnnovak made? The hardware ones that is.
> 
> Nothing. I haven't recorded the NukedSC55 versions yet, that is all. Just SCVA and real hardware (SC-55 mk1).
> 
> EDIT: Oh, okay, I see @markanini is talking about getting the real hardware recordings from vgmpf.com. Well, you can do that too. But better yet, use my REAPER projects to send the MIDI data to NukedSC55 via virtual MIDI cables using my own 46 game MIDI captures.

I'm confused once more. Are your recordings not hardware recordings? Did you not record audio from a real sc-55?

I'm guessing there I some terminology here that I don't understand.

**Comment by markanini** [comment]
> > > I'm confused, what is wrong with the recordings johnnovak made? The hardware ones that is.
> > 
> > 
> > Nothing. I haven't recorded the NukedSC55 versions yet, that is all. Just SCVA and real hardware (SC-55 mk1).
> > EDIT: Oh, okay, I see @markanini is talking about getting the real hardware recordings from vgmpf.com. Well, you can do that too. But better yet, use my REAPER projects to send the MIDI data to NukedSC55 via virtual MIDI cables using my own 46 game MIDI captures.
> 
> I'm confused once more. Are your recordings not hardware recordings? Did you not record audio from a real sc-55?
> 
> I'm guessing there I some terminology here that I don't understand.

I'm saying that johnnovaks comparison page doesn't include Nuked SC55 at this time, and sharing the basic steps in making your own comparison.

**Comment by johnnovak** [comment]
> I'm confused once more. Are your recordings not hardware recordings? Did you not record audio from a real sc-55?

Read my article, man :)

**Comment by Karmeck** [comment]
> There are far too many comments here and you should all stop arguing.
> 
> I am curious however. In my and some others' experience, the early SC DACs have a weird aliasing-related behaviour; all the frequencies below 32kHz get reflected above 32kHz. I'm not sure how high they get reflected to but it's easy to see on a 44.1kHz recording. That affects the sound of them, giving a bit of extra brightness, though it's fairly subtle. Could this be emulated? This is probably not a complicated effect to reproduce.

I wonder if this I related to this issue:
https://www.vogons.org/viewtopic.php?p=1261758#p1261758

**Comment by johnnovak** [comment]
> all the frequencies below 32kHz get reflected above 32kHz. I'm not sure how high they get reflected to but it's easy to see on a 44.1kHz recording.

Aliasing is when frequencies above half the Nyquist frequency (half the sample rate) get folded back to the audible range, for a variety of reasons. It can happen at the ADC stage, DAC stage, sample interpolation stage, DSP calculations, etc.

In this case, all sorts of junk above 16kHz will get folded back. I tried to find some videos online that illustrate this with a sweep, but couldn't and all textual descriptions are vague...

If you play some SC-55 patch in the high octaves, and you use the pitch wheel, and you examine the output in a spectrum analyser, you'll see the actual note you are playing go up as you bend up, but there will be some junk at much lower frequencies moving in the _opposite_ direction, so downwards! 

Yeah aliasing can add some brightness because of added quasi-random freqs, but the problem is those extra freqs are inharmonic, so generally it sounds ugly... especially with pitch bends, vibrato, etc.

But it can sound cool with drums.

**Comment by markanini** [comment]
Am I the only one who thinks the output of Nuked SC55 sounds punchier and better defined than any analogue capture of a real SC-55? I for one don't miss anything. Just my two cents.

**Comment by johnnovak** [comment]
> Am I the only one who thinks the output of Nuked SC55 sounds punchier and better defined than any analogue capture of a real SC-55? I for one don't miss anything. Just my two cents.

A/B compare the System Shock intro tune with exactly matched volumes, then report back again 😄

**Comment by markanini** [comment]
> A/B compare the System Shock intro tune with exactly matched volumes, then report back again 😄

I did with this as my point of comparison https://www.youtube.com/watch?v=DENSsyd2igY and this for the midi files https://www.systemshock.org/index.php?topic=2285.0 The snare was obviously different so I tried using the capital fallback feature in FSMP and it worked to match the instrument sounds. No matter what I think it's likely better to discuss midi music that doesn't have issues with sounds being changed between different SC55 versions in the first place. (Nuked SC55 sounded better to me)

**Comment by johnnovak** [comment]
Yeah, so that screenshot in the first post must be the DACs output prior to the analog reconstruction filter. You'd need to repeat the measurement with the actual analog output jacks.

https://www.sciencedirect.com/topics/engineering/reconstruction-filter

Relevant quote:

> The output from the DAC will be ‘steppy’ because the DAC can only output certain voltage values. For example, an 8-bit DAC will have 256 different output voltage levels going from perhaps −5 V to +5 V. When this quantized output is analysed, frequency components of fs, 2fs, 3fs, 4fs etc. (harmonics of the sampling frequency) are found. The very action of sampling and converting introduces these harmonics of the sampling frequency into the output signal. It is these harmonics which give the output signal its steppy appearance. The reconstruction filter is a [lowpass filter](https://www.sciencedirect.com/topics/engineering/lowpass-filter) having a cut-off frequency of fs/2, and is used to filter out these harmonics and so smooth the output signal.

I'm certain the _real_ final output is not steppy; that would sound like crap...

> The output of a DAC is not a continuously varying waveform, but instead a series of DC levels. This output must be passed through a filter to remove the [high frequency components](https://www.sciencedirect.com/topics/engineering/high-frequency-component) and smooth waveform into a more truly [analog waveform](https://www.sciencedirect.com/topics/engineering/analog-waveform).

Seems like the original poster is a bit confused about how DACs operate.

**Comment by nikitalita** [comment]
> Yeah, so that screenshot in the first post must be the DACs output prior to the analog reconstruction filter. You'd need to repeat the measurement with the actual analog output jacks.

I did. See here: https://github.com/nukeykt/Nuked-SC55/issues/50#issuecomment-2067927736

**Comment by johnnovak** [comment]
> > Yeah, so that screenshot in the first post must be the DACs output prior to the analog reconstruction filter. You'd need to repeat the measurement with the actual analog output jacks.
> 
> I did. See here: [#50 (comment)](https://github.com/nukeykt/Nuked-SC55/issues/50#issuecomment-2067927736)

That's the end of it, then 😆
